### PR TITLE
added and tested drop

### PR DESCRIPTION
**Pull request by MatthewChau99**
Please take a look :)

### CODE REVIEW

File: test/pipeline_test.py [hunk 1]
```diff
@@ -15,5 +17,42 @@ def _test():
...
+def test_drop():
+    pandas_df = pd.DataFrame(data={
+        'col1': [0, 1, 2, 3],
+        'col2': [3, 4, 5, 6],
+        'col3': [6, 7, 8, 9],
+        'col4': [9, 10, 11, 12]
+    })
+
+    # Drop by columns
+    try:
```
Comment (ccharp) [inline test/pipeline_test.py:29]: Are the try/except blocks necessary? I we'd just let the exception `pd.testing.assert_frame_equal` bubble up.

File: test/pipeline_test.py [hunk 2]
```diff
@@ -15,5 +17,42 @@ def _test():
...
+        df1 = DplyFrame(pandas_df)
+        output1 = df1 + drop(['col3', 'col4'], axis=1)
+        expected1 = pandas_df.drop(['col3', 'col4'], axis=1)
+        pd.testing.assert_frame_equal(output1.pandas_df, expected1)
+
+        print(pandas_df)
```
Comment (ccharp) [inline test/pipeline_test.py:35]: Remove extraneous print

File: src/pipeline.py [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+def query(query_str: str):  # summary_type: e.g. mean, variance, other descriptive stats, num rows,
     return lambda d1: DplyFrame(d1.pandas_df.query(query_str))
 
-def apply(func, axis=0, **kwargs): # TODO: CSC: use kwargs... couldn't quickly figure out how to do it
-    return lambda d1: DplyFrame(d1.pandas_df.apply(func=func, axis=axis)) 
+
+def apply(func, axis=0, **kwargs):  # TODO: CSC: use kwargs... couldn't quickly figure out how to do it
+    return lambda d1: DplyFrame(d1.pandas_df.apply(func=func, axis=axis))
+
+
```
Comment (ccharp) [inline src/pipeline.py:11]: Remove extra new line

File: src/pipeline.py [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+from src.dplypy import DplyFrame
 
-def query(query_str: str): # summary_type: e.g. mean, variance, other descriptive stats, num rows, 
+
+def query(query_str: str):  # summary_type: e.g. mean, variance, other descriptive stats, num rows,
     return lambda d1: DplyFrame(d1.pandas_df.query(query_str))
 
-def apply(func, axis=0, **kwargs): # TODO: CSC: use kwargs... couldn't quickly figure out how to do it
-    return lambda d1: DplyFrame(d1.pandas_df.apply(func=func, axis=axis)) 
+
```
Comment (ccharp) [inline src/pipeline.py:7]: Remove extra new line

File: src/pipeline.py [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
+def apply(func, axis=0, **kwargs):  # TODO: CSC: use kwargs... couldn't quickly figure out how to do it
+    return lambda d1: DplyFrame(d1.pandas_df.apply(func=func, axis=axis))
+
+
+def drop(labels=None, axis=0, index=None, columns=None):
```
Comment (ccharp) [inline src/pipeline.py:12]: I know this hasn't been done yet, but let's get in the habit writing API documentation for the function as it's implemented.

File: test/pipeline_test.py [hunk 6]
```diff
@@ -15,5 +17,42 @@ def _test():
...
+        output3 = df3 + drop([0, 1])
+        expected3 = pandas_df.drop([0, 1])
+        pd.testing.assert_frame_equal(output3.pandas_df, expected3)
+
+        df4 = DplyFrame(pandas_df)
+        output4 = df4 + drop(index=[2, 3])
+        expected4 = pandas_df.drop(index=[2, 3])
+        pd.testing.assert_frame_equal(output4.pandas_df, expected4)
+    except:
+        raise
```
Comment (ccharp) [inline test/pipeline_test.py:55]: We may also consider testing error reporting. If the caller passes in bad input, for example, that causes Pandas to throw and exception. Should we just let the Pandas exception bubble up? I think yes, unless we have a compelling reason not to.

**Review by ccharp — CHANGES_REQUESTED** [review]

File: test/pipeline_test.py (hunk 1 above)
Comment (MatthewChau99) [reply test/pipeline_test.py]: I think you're right. Those try except are not necessary.

File: src/pipeline.py (hunk 3 above)
Comment (MatthewChau99) [reply src/pipeline.py:11]: I looked up PEP8 and found that top-level functions are separated by two blank lines, and class methods are separated by one blank line. Perhaps it's better to leave it as two?

Comment (ccharp) [reply src/pipeline.py:11]: I see, I see. Yeah, let's follow the PEP8 guidelines. Maybe we run linter each as a pre-commit hook?

**Review by ccharp — APPROVED** [review]

File: src/pipeline.py (hunk 5 above)
Comment (MatthewChau99) [reply src/pipeline.py:12]: Got it! Updated in my next commit. Do we need to write documentation for tests as well?

**Review by nickjsr — APPROVED** [review]

**Pull request merged**